### PR TITLE
Fix Environment.OSVersion.Platform output on MacOSX

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/19694

### CODE REVIEW

**Comment by danmosemsft** [comment]
test?

**Comment by justinvp** [comment]
I thought returning PlatformId.Unix on macOS was by design to match Mono's behavior.

#11878
https://github.com/terrajobst/platform-compat/issues/29

**Comment by stephentoub** [comment]
It is by design to match Mono.

**Comment by danmosemsft** [comment]
That's fine, but @ViktorHofer could you please then add a comment to the code instead? Otherwise tihs will come up agian.

**Comment by ViktorHofer** [comment]
@danmosemsft yes but I will wait till we decided if we should also add annotations. See https://github.com/dotnet/corefx/issues/19694.

**Comment by ViktorHofer** [comment]
@jkotas we aren't changing it. This PR is halted till the decision is made here: https://github.com/dotnet/corefx/issues/19694

**Comment by ViktorHofer** [comment]
@jkotas @stephentoub PTAL

File: src/System.Runtime.Extensions/src/System/PlatformID.cs [hunk 1]
```diff
@@ -2,16 +2,18 @@
...
 
+using System.ComponentModel;
+
 namespace System
 {
     public enum PlatformID
     {
-        Win32S = 0,
-        Win32Windows = 1,
+        [EditorBrowsable(EditorBrowsableState.Never)] Win32S = 0,
```
Comment (jkotas) [inline src/System.Runtime.Extensions/src/System/PlatformID.cs]: It is most important to have these annotations in the reference assembly `src/System.Runtime.Extensions/ref/...`. The editor looks for the attributes in the reference assembly.

Annotating implementation is fine too.

Comment (ViktorHofer) [inline src/System.Runtime.Extensions/src/System/PlatformID.cs]: done

**Pull request merged**